### ISSUE TITLE
Exercise 4 k-NN (with Baseline)

### ISSUE THREAD

**Issue by orlandocosta76** [LDSSA/batch2-blu12#3]
As in issue #2, this exercise has the same problems in the asserts:
- uses list instead of set
- parameter  learning rate of 0.0005 is compared with 5.-e5 instead of 5.-e4

PS-I propose to replace those asserts with:
```
assert(set(baseline.bsl_options.values()) == {'sgd', 5e-04, 0.05})
assert(set(baseline_results.keys()) == {'test_rmse', 'test_mae', 'fit_time', 'test_time'})
```

**Comment by hcastilho** [comment]
I will have to check if the grader will let me change this without breaking anything.
In the meantime put the options in this order:
bsl_options: 'method', 'learning_rate', 'reg'
sim_options: 'name', 'user_based', 'min_support'
As the grader is using python 3.6 this will work.

**Comment by orlandocosta76** [comment]
Ok @hcastilho  I did that for both issues and it worked, thanks.
